product_taxes_group/model/tax_group.py

```diff
@@ -0,0 +1,204 @@
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import logging
+
+from openerp import SUPERUSER_ID, models, fields, api, _
+from openerp.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class TaxGroup(models.Model):
+    """Group of customer and supplier taxes.
+    This group is linked to a product to select a group of taxes in one
+    time."""
+    _name = 'tax.group'
+    _description = 'Taxes Group'
+    _MAX_LENGTH_NAME = 256
+
+    # Getter / Setter Section
+    def _default_company_id(self):
+        return self.env['res.users']._get_company()
+
+    def _get_product_tmpl_qty(self):
+        for rec in self:
+            rec.product_tmpl_qty = self.env['product.template'].search_count([
+                ('tax_group_id', '=', rec.id), '|', ('active', '=', False),
+                ('active', '=', True)])
+
+    def _get_product_tmpl_ids(self):
+        for rec in self:
+            rec.product_tmpl_ids = self.env['product.template'].search([
+                ('tax_group_id', '=', rec.id), '|', ('active', '=', False),
+                ('active', '=', True)])
+
+    # Field Section
+    name = fields.Char(
+        size=_MAX_LENGTH_NAME, required=True, select=True)
+
+    company_id = fields.Many2one(
+        comodel_name='res.company', default=_default_company_id,
+        string='Company', help="Specify a company"
+        " if you want to define this Taxes Group only for specific company."
+        " Otherwise, this Taxes Group will be available for all companies.")
+
+    active = fields.Boolean(
+        default=True,
+        help="If unchecked, it will allow you to hide the Taxes"
+        " Group without removing it.")
+
+    product_tmpl_ids = fields.One2many(
+        comodel_name='product.template', string='Products',
+        compute=_get_product_tmpl_ids)
+
+    product_tmpl_qty = fields.Integer(
+        string='Products Quantity', compute=_get_product_tmpl_qty)
+
+    supplier_tax_ids = fields.Many2many(
+        'account.tax', 'product_supplier_tax_rel',
+        'group_id', 'tax_id', string='Supplier Taxes', domain="""[
+            ('company_id', '=', company_id),
+            ('parent_id', '=', False),
+            ('type_tax_use', 'in', ['purchase', 'all'])]""")
+
```

Comment:
The old ones are: 'sale_base_tax_ids' and 'purchase_base_tax_ids'.
And also think it needs to change the relation table names.